specs/CustomDownload.md

```diff
@@ -0,0 +1,775 @@
+            MessageBox.Show(this, "Connecton with file host broken.", "Download Interrupted");
+            if (download.CanResume == true)
+            {
+                download.Resume();
+            }
+            break;
+        case CoreWebView2DownloadState.DownloadComplete:
+            break;
+        }
+    };
+}
+```
+# Remarks
+<!-- TEMPLATE
+    Explanation and guidance that doesn't fit into the Examples section.
+
+    APIs should only throw exceptions in exceptional conditions; basically,
+    only when there's a bug in the caller, such as argument exception.  But if for some
+    reason it's necessary for a caller to catch an exception from an API, call that
+    out with an explanation either here or in the Examples
+-->
+
+
+# API Notes
+<!-- TEMPLATE
+    Option 1: Give a one or two line description of each API (type and member),
+        or at least the ones that aren't obvious from their name. These
+        descriptions are what show up in IntelliSense. For properties, specify
+        the default value of the property if it isn't the type's default (for
+        example an int-typed property that doesn't default to zero.)
+
+    Option 2: Put these descriptions in the below API Details section,
+        with a "///" comment above the member or type.
+-->
+See [API Details](#api-details) section below for API reference.
+
+# API Details
+<!-- TEMPLATE
+    The exact API, in IDL format for our COM API and
+    in MIDL3 format (https://docs.microsoft.com/en-us/uwp/midl-3/)
+    when possible, or in C# if starting with an API sketch for our .NET and WinRT API.
+
+    Include every new or modified type but use // ... to remove any methods,
+    properties, or events that are unchanged.
+
+    (GitHub's markdown syntax formatter does not (yet) know about MIDL3, so
+    use ```c# instead even when writing MIDL3.)
+
+    Example:
+
+    ```
+    /// Event args for the NewWindowRequested event. The event is fired when content
+    /// inside webview requested to open a new window (through window.open() and so on.)
+    [uuid(34acb11c-fc37-4418-9132-f9c21d1eafb9), object, pointer_default(unique)]
+    interface ICoreWebView2NewWindowRequestedEventArgs : IUnknown
+    {
+        // ...
+
+        /// Window features specified by the window.open call.
+        /// These features can be considered for positioning and sizing of
+        /// new webview windows.
+        [propget] HRESULT WindowFeatures([out, retval] ICoreWebView2WindowFeatures** windowFeatures);
+    }
+    ```
+
+    ```c# (but really MIDL3)
+    public class CoreWebView2NewWindowRequestedEventArgs
+    {
+        // ...
+
+	       public CoreWebView2WindowFeatures WindowFeatures { get; }
+    }
+    ```
+-->
+## Win32 C++
+```c#
+// Enums and structs
+
+typedef enum COREWEBVIEW2_DOWNLOAD_STATE {
+  /// The download is in progress. Includes downloads paused by user.
+  DOWNLOAD_IN_PROGRESS,
```

Comment:
In COM our enums all start with the name of the enum like COREWEBVIEW2_DOWNLOAD_STATE_*. In .NET and WinRT this isn't the case and our generated code will remove that prefix along with changing the casing and removing _.

So here it could be COREWEBVIEW2_DOWNLOAD_STATE_IN_PROGRESS which would become InProgress in .NET & WinRT.